pkg/controller/jenkins/configuration/base/reconcile.go

```diff
@@ -1092,3 +1133,39 @@ func (r *ReconcileJenkinsBaseConfiguration) handleAdmissionControllerChanges(cur
 		}
 	}
 }
+
+func verifyAPI(group string, version string) (bool, error) {
+	cfg, err := config.GetConfig()
+	if err != nil {
+		// Unable to get KBS Config
+		return false, err
+	}
+
+	k8s, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		// Unable to create K8s client
+		return false, err
+	}
+
+	gv := schema.GroupVersion{
+		Group:   group,
+		Version: version,
+	}
+
+	if err = discovery.ServerSupportsVersion(k8s, gv); err != nil {
+		// error, API not available
+		return false, nil
+	}
+
+	// API Exists
+	return true, nil
+}
+
+func verifyRouteAPI() error {
+	found, err := verifyAPI(routev1.GroupName, routev1.SchemeGroupVersion.Version)
+	if err != nil {
+		return err
+	}
+	routeAPIFound = found
```

Comment:
Is this needed?

Reply:
It will be used to check if the route api exists